src/ORM/DataObject.php

```diff
@@ -3755,7 +3789,7 @@ public function searchableFields()
 
                     if ($schema->fieldSpec($this, $name)) {
                         $fields[] = $name;
-                    } elseif ($this->relObject($spec)) {
+                    } elseif ($this->relObject($spec) && $this->hasRelations($spec)) {
```

Comment:
Under what circumstances does `relObject()` return a value record, but `hasRelations()` returns false?
The PHPDoc for `relObject` says:
> If a relation is blank, this will return null instead.
> If a relation name is invalid (e.g. non-relation on a parent) this
> can throw a LogicException.

In other words, `relObject()` should _always_ return null if the relation doesn't exist.

Reply:
I have excluded fields like `CMSThumbnail` and `StripThumbnail` from `$searchable_fields`. Since, for example `ImageManipulation` adds these methods to the `File` class and they returns `DBHTMLText` they are not part of any relations and don't have any fields in DB. In the `relObject`  method, they easily passed the hasMethod check and are added to the search fields accordingly, but since they are not fields in the data base, they fail the request.
I think we mention about this issue [here](https://docs.silverstripe.org/en/4/developer_guides/model/scaffolding/#searchable-fields)

> $searchable_fields will default to use the [$summary_fields config](https://docs.silverstripe.org/en/4/developer_guides/model/scaffolding/#summary-fields) if not defined. This works fine unless your $summary_fields config specifies fields that are not stored in the database.

I don't want to add this checking in `relObject` since we use this method in another places, not for `searchable_fields`, probably for some logic it's acceptable to get `$relation` that passed `hasMethod` check.